presto-main/src/main/java/com/facebook/presto/sql/planner/DomainTranslator.java

```diff
@@ -113,30 +118,30 @@ public Expression toPredicate(TupleDomain<Symbol> tupleDomain)
                 .collect(collectingAndThen(toImmutableList(), ExpressionUtils::combineConjuncts));
     }
 
-    private Expression toSymbolExpression(Symbol symbol)
+    private static Expression toSymbolExpression(Symbol symbol)
     {
         if (symbol instanceof SymbolWithSubfieldPath) {
-            SymbolWithSubfieldPath subfieldPath = (SymbolWithSubfieldPath) symbol;
-            SubfieldPath path = subfieldPath.getPath();
-            Expression base = new SymbolReference(path.getPath().get(0).getField());
+            SubfieldPath path = ((SymbolWithSubfieldPath) symbol).getPath();
+            Expression base = new SymbolReference(symbol.getName());
             for (int i = 1; i < path.getPath().size(); i++) {
                 SubfieldPath.PathElement element = path.getPath().get(i);
-                String field = element.getField();
-                if (element.getIsSubscript()) {
-                    base = new SubscriptExpression(base, field != null ? new StringLiteral(field) : new LongLiteral(Long.valueOf(element.getSubscript()).toString()));
+                if (element instanceof NestedField) {
+                    base = new DereferenceExpression(base, new Identifier(((NestedField) element).getName()));
                 }
-                else if (field != null) {
-                    base = new DereferenceExpression(base, new Identifier(field));
+                else if (element instanceof LongSubscript) {
+                    base = new SubscriptExpression(base, new LongLiteral(String.valueOf(((LongSubscript) element).getIndex())));
```

Comment:
I think there's some extra parenthesis here and below

Reply:
@tdcmeehan I don't see that. Would you show a rewrite?

Reply:
You're right, I didn't read it correctly.